### ISSUE TITLE
Pulse Sequencer Display

### ISSUE THREAD

**Issue by bhemmerling** [HaeffnerLab/Haeffner-Lab-LabRAD-Tools#37]
Once a pulse sequence is started in script scanner, it would be nice if the actual sequence, i.e. pulses in each channel vs. tim, was plotted in a separate tab. Could help the user to find errors in the sequence more easily.

**Comment by jayich** [comment]
Depending on what the sequence is doing it might be better to have a more abstract representation, as compared to a plot.  For example you don't want to plot sequences that have events on the microsecond time scale with events on the millisecond time scale.  I have been thinking about making a control that would allow changing event sequences, etc.  The interface would basically look like the Cicero control interface.  If this was also used as an indicator for abstracted pulse sequencing this would be a nice interface.  In regards to the "control" aspect I thought having an experiment that runs whatever sequence is represented by the GUI interface would allow for a different control knob compared to the more abstract sequencing.

There is interest in this type of interface from several people.

**Comment by djgorman** [comment]
Yeah my plan for this is to be able to plot either the whole sequence or be able to select by subsequences.

As far as making a more abstract interface for pulse sequencing... I agree that we should abstract this a more, but that's more of a medium-term goal. Right now my basic plan is an interface which can show either the entire pulse sequence or some subsequence. I also want to select which channels to display. I'm probably going to use PyQtGraph because it's nicely integrated into Qt. It may be possible to extend that thing into a framework for designing pulse sequences but I haven't really thought about the best way to do that exactly.

**Comment by aransfor** [comment]
:8ball:
